VMware.CloudFoundation.CertificateManagement.psm1

```diff
@@ -25,17 +25,924 @@ if ($PSEdition -eq 'Desktop') {
+
+
+Function Get-CertManagementModeForESXi {
+    <#
+        .SYNOPSIS
+        Retrieves the certificate management mode value from the vcenter server for the given domain.
+
+        .DESCRIPTION
+        Get-CertManagementModeForESXi cmdlet retrieves the cert management mode value from vcenter for the given domain.
+
+        .EXAMPLE
+        Get-CertManagementModeForESXi -server sfo-vcf01.sfo.rainpole.io -user administrator@vsphere.local -pass VMw@re123! -domain sfo-m01
+        This example retrieves the certificate management mode value for the vcenter server belonging to domain sfo-m01.
+
+    #>
+
+    Param (
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $server,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $user,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $pass,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $domain
+    )
+    
+    Try {
+        $vcfVcenterDetails = Get-vCenterServerConnection -server $server -user $user -pass $pass -domain $domain
+        $entity = Connect-VIServer -Server $vcfVcenterDetails.fqdn -User $vcfVcenterDetails.ssoAdmin -Pass $vcfVcenterDetails.ssoAdminPass
+        $certModeSetting = Get-AdvancedSetting "vpxd.certmgmt.mode" -Entity $entity
+        return $certModeSetting.value
+    }  
+    Catch {
+        Debug-ExceptionWriter -object $_
+    }
+    Finally {
+        Disconnect-VIServer $vcfVcenterDetails.fqdn -Confirm:$false -WarningAction SilentlyContinue
+    }
+}    
+
+Export-ModuleMember -Function Get-CertManagementModeForESXi
+
+Function Set-CertManagementModeForESXi {
+
+    <#
+        .SYNOPSIS
+        Sets the ESXi host's management mode in the vCenter Server to either custom or vmca.
+
+        .DESCRIPTION
+        Set-CertManagementModeForESXi cmdlet sets the ESXi host's management mode on the vCenter server belonging to given domain.
+
+        .EXAMPLE
+        Set-CertManagementModeForESXi -server sfo-vcf01.sfo.rainpole.io -user administrator@vsphere.local -pass VMw@re123! -domain sfo-m01 -mode custom
+        This example sets the ESXi management mode to custom for the vCenter Server belonging to domain sfo-m01
+    #>
+
+    Param (
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $server,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $user,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $pass,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $domain,
+        [Parameter (Mandatory = $true)] [ValidateSet ("custom", "vmca")] [String] $mode
+
+    )
+    Try {
+        $vcfVcenterDetails = Get-vCenterServerConnection -server $server -user $user -pass $pass -domain $domain
+        $entity = Connect-VIServer -Server $vcfVcenterDetails.fqdn -User $vcfVcenterDetails.ssoAdmin -Pass $vcfVcenterDetails.ssoAdminPass
```

Comment:
Doesn't the `Get-vCenterServerConnection` function already connect to the vCenter Server instance?

Reply:
I will verify this functionality. Keeping it open.

Reply:
Re-wrote the Get-vCenterServerConnection to just Get-vCenterServer which returns a single object with details and connection.

Reply:
O@  One thing that it just made me think is that it could be `Connect-vCenterServer` if it might be more appropriate. Up to you!